frappe/model/base_document.py

```diff
@@ -225,12 +225,13 @@ def _init_child(self, value, key):
 		if not self.doctype:
 			return value
 
-		if not isinstance(value, BaseDocument):
-			value["doctype"] = self.get_table_field_doctype(key)
-			if not value["doctype"]:
-				raise AttributeError(key)
+		doctype = self.get_table_field_doctype(key)
+		if not doctype:
+			return value
```

Comment:
This can alternatively be put inside the `if not isinstance(value, BaseDocument):` block below.